os/x11/window.cpp

```diff
@@ -1042,6 +1042,20 @@ void WindowX11::processX11Event(XEvent& event)
         KEY_TRACE("Xutf8LookupString %s\n", &buf[0]);
       }
 
+      // Check if key has been pressed, and then: 1. check if
+      // previous event was a key press, 2. if the key from previous
+      // event is still pressed, 3. if the key from previous event
+      // has the same symbolic name as the previous key. If yes,
+      // then it means this button is being held down
+      XEvent prev_event = checkPrevKeyEvent();
+      if (prev_event.type != 0 && event.type == KeyPress) {
+        KeySym prev_keysym = XLookupKeysym(&prev_event.xkey, 0);
+        if (prev_event.type == KeyPress && x11_is_key_pressed(x11_keysym_to_scancode(prev_keysym)) && prev_keysym == keysym)
+          ev.setRepeat(1);
+      }
+
+      setPrevKeyEvent(event);
```

Comment:
I think we could store the previous pressed KeySym instead of the XEvent and then refactor the code a bit:
 
```suggestion
      if (event.type == KeyPress) {
        if (m_pressedKeySym == keysym)
          ev.setRepeat(1);
        m_pressedKeySym = keysym;
      }
      else 
        m_pressedKeySym = 0;
```

Reply:
Done as suggested, right - it will simplify logic a lot.